cdc/processor/sinkmanager/mem_quota.go

```diff
@@ -156,6 +160,30 @@ func (m *memQuota) release(tableID model.TableID, resolved model.ResolvedTs) {
 	}
 }
 
+// clean all records of the table.
+func (m *memQuota) clean(tableID model.TableID) {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+
+	if _, ok := m.tableMemory[tableID]; !ok {
+		// This can happen when the table has no data and never been recorded.
+		log.Warn("Table consumed memory records not found.",
+			zap.String("namespace", m.changefeedID.Namespace),
+			zap.String("changefeed", m.changefeedID.ID),
+			zap.Int64("tableID", tableID))
+		return
+	}
+	records := m.tableMemory[tableID]
+	for _, record := range records {
+		m.usedBytes -= record.size
+	}
+	delete(m.tableMemory, tableID)
+
+	if m.usedBytes < m.totalBytes {
+		m.blockAcquireCond.Signal()
```

Comment:
Why not awake all here? So the another `awake` isn't necessary.

Reply:
I think we have to stop completely before we can proceed to clean.